src/spec-lang/tc/typecheck.ts

```diff
@@ -731,26 +796,32 @@ function tcIdVariable(expr: SId, ctx: STypingCtx, typeMap: TypeMap): SType | und
         return astVarToSType(def);
     }
 
-    const [letNode, bindingIdx] = def;
-    const rhsT = tc(letNode.rhs, ctx, typeMap);
+    const [node, bindingIdx] = def;
 
-    if (letNode.lhs.length > 1) {
-        if (!(rhsT instanceof STupleType && rhsT.elements.length === letNode.lhs.length)) {
-            throw new SExprCountMismatch(
-                `Wrong number of values for let bindings in ${letNode.pp()}. Expected ${
-                    letNode.lhs.length
-                } values, instead got ${rhsT.pp()}`,
-                letNode
-            );
+    if (node instanceof SLet) {
```

Comment:
Can we reduce nesting here by using early `return`s?

Reply:
Fixed in 031fe1c.